enhancements/builds/entitlements.md

```diff
@@ -0,0 +1,387 @@
+>an upstream features becomes more likely by the time implementation of this proposal arrives, it might dictate our 
+>choosing the CRI-O related path vs. building a path around post-install Secrets, as that later path would
+>most likely be obviated by the upstream feature.
+> 3. We did provide for multiple sets of entitlement credentials in 3.x.  There have been no official, RFE style  
+>registered requirements entitlements at a namespace level.  However, we've started to get (slightly) less official
+>feedback that customers want this in a more automated sense (vs. the current manual steps).  At a minimum, and is
+>noted below when alternatives are discussed, if namespaced entitlements are added with subsequent implementation
+>after global entitlements usability is improved, it needs to be seamless, and we minimally need to agree now
+>on what the order of precedence will be (with the take on that listed below).
+> 4. There has also been a question in the past needing to shard entitlements due to API rate limiting.  To date, it
+>has been stated that is not a concern for builds.  Periodic checkpoints here could be prudent.
+> 5. There is also the general question of how entitlements are propagated for our managed service offerings?  Reaching 
+>out to the appropriate teams there to understand current capabilities/processes and building from there is needed.
+
+## Summary
+
+So in V3.x, all OpenShift Builds essentially had the customer RHEL entitlements mounted into the build pod automatically
+and without further intervention after install "for free" because:
+
+ - the ansible based installer required/forced providing your subscription information in order to install OCP
+ - for reference, see [these docs](https://docs.openshift.com/container-platform/3.11/install/host_preparation.html#host-registration) 
+ and [these roles](https://github.com/openshift/openshift-ansible/blob/release-3.11/roles/rhel_subscribe/tasks/main.yml)
+ - next, Red Hats's docker daemon installed in 3.x had specific logic that auto-mounted the host's subscriptions 
+ into the container
+
+In V4.x, with 
+
+- the removal of the docker daemon, meaning no docker socket and no access to the host's content
+- and the total replacement of the openshift ansible based installer, so no gathering of entitlements during install
+
+all those 3.x mechanisms are gone, and a series of steps from either the cluster admin and/or developer attempting
+to run OpenShift Builds leveraging entitled content must be performed after install to enable access to the entitled
+content during the build.
+
+The highlights of those steps from [the current 4.x doc](https://docs.openshift.com/container-platform/4.3/builds/running-entitled-builds.html):
+
+1) First, either
+ 
+- create an ImageStreamTag to the UBI image from `registry.redhat.io` in the `openshift` namespace, or 
+- reference the UBI image directly and include a pull secret to `registry.redhat.io` in your BuildConfig
+
```

Comment:
When using ImageStreams we can use the ImageChange trigger for builds. I do not know if we should document this here or where, but customers were confused why would I use an ImageStream when I can reference it directly from `registry.redhat.io`.

Reply:
Yeah using imagestreams here is no different that using imagestreams for other scenarios.

There are several reasons to use imagestreams in a general sense.

But one advantage wrt `registry.redhat.io` is the `openshift` namespace is already seeded with credentials for `registry.redhat.io`, and so there is no need to mount those credentials into the build.

And yep, the current doc link at https://docs.openshift.com/container-platform/4.3/builds/running-entitled-builds.html#builds-create-imagestreamtag_running-entitled-builds does not clearly articulate that advantage when discussing creating it in a user namespace vs. the openshift namespace ... it just mentions making it accessible to all users.

I've opened https://bugzilla.redhat.com/show_bug.cgi?id=1806755 for the docs team to clear that up.

Reply:
I should also note that my teammates @ricardomaraschini and @dmage just dropped a feature in the image registry operator that mounts the install pull secret in the image registry as part of 4.5 dev.

Now *any* imagestream, local namespace or openshift namespace, will be able to pull from `registry.redhat.io` automagically.  So another advantage for using imagestreams, vs. having to mount the secret into the specific buildconfig so buildah can pull direct references during the build.

Reply:
but we're also going to make builds use that pull-secret directly, so once that is done, builds will be able to directly pull from registry.redhat.io even when not using an imagestream.

Reply:
ah ha ... I missed that req coming down the pipe ... thanks for the enlightenment @bparees 

bottom line ... some adding in this proposal some references around these niche improvements coming from other feature seems good .... I'll add that to the list.

Reply:
ok reference added and will be part of next push